src/Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -141,8 +155,15 @@ protected override async Task WriteCacheBytesAsync(string cacheKey, byte[] bytes
             _logger.LogDebug($"[IdWebCache] MemoryCache: Write cacheKey {cacheKey} Byte size: {bytes?.Length} Time in Ticks: {Utility.Watch.Elapsed.Ticks - startTicks}. ");
             _logger.LogDebug($"[IdWebCache] MemoryCache: Count: {_memoryCache.Count}");
 
-            await _distributedCache.SetAsync(cacheKey, bytes, _distributedCacheOptions).ConfigureAwait(false);
-            _logger.LogDebug($"[IdWebCache] DistributedCache: Write cacheKey {cacheKey} Byte size {bytes?.Length} Time in Ticks: {Utility.Watch.Elapsed.Ticks - startTicks}. ");
+            try
+            {
+                await _distributedCache.SetAsync(cacheKey, bytes, _distributedCacheOptions).ConfigureAwait(false);
```

Comment:
I don't think we should await this one, we should let it execute async, and we can also handle the exception async as well. We should treat L2 as optimistic - delays to the incoming call is too big.

Reply:
By not awaiting, do we have a risk of not respecting causality ?

Reply:
I'd like to discuss this one more. I'll merge this PR and then open another w/this change.